pyscriptjs/src/pyodide.ts

```diff
@@ -1,4 +1,84 @@
-import type { loadPyodide } from 'pyodide';
+import { Runtime } from './runtime';
+import { getLastPath, inJest } from './utils';
+import type { PyodideInterface } from './runtime';
+import { loadPyodide } from 'pyodide';
+// eslint-disable-next-line
+// @ts-ignore
+import pyscript from './python/pyscript.py';
 
-// The current release doesn't export `PyodideInterface` type
-export type PyodideInterface = Awaited<ReturnType<typeof loadPyodide>>;
+export class PyodideRuntime extends Runtime {
+    src = 'https://cdn.jsdelivr.net/pyodide/v0.21.1/full/pyodide.js';
+    name = 'pyodide-default';
+    lang = 'python';
+    interpreter: PyodideInterface;
+    globals: any;
+
+    async loadInterpreter(): Promise<void> {
+        console.log('creating pyodide runtime');
+        // eslint-disable-next-line
+        // @ts-ignore
+        let extraOpts: any = {}
+        if (inJest()) {
+          extraOpts = {indexURL: [process.cwd(), 'node_modules', 'pyodide'].join('/') }
+	}
+        this.interpreter = await loadPyodide({
+            stdout: console.log,
+            stderr: console.log,
+            fullStdLib: false,
+            ...extraOpts
+        });
+
+        this.globals = this.interpreter.globals;
+
+        // now that we loaded, add additional convenience functions
+        console.log('loading micropip');
+        await this.loadPackage('micropip');
+
+        console.log('loading pyscript...');
+        const output = await this.run(pyscript);
+        if (output !== undefined) {
+            console.log(output);
+        }
+
+        console.log('done setting up environment');
+    }
+
+    async run(code: string): Promise<any> {
+        return await this.interpreter.runPythonAsync(code);
+    }
+
+    registerJsModule(name: string, module: object): void {
+        this.interpreter.registerJsModule(name, module);
```

Comment:
Almost feels like it's an unnecessary double layer of abstraction. If `Runtime` already allows for th injection of a runtime through `interpreter`, do we need to also subclass and reimplement these methods here?

This PR is a good first step towards building a common interface, so we don't need to figure this out in this context but we can think about this after we merge.

Reply:
noted.